### PR TITLE
Add gRPC configuration parity across JVM and Native

### PR DESCRIPTION
**Pull request by nickjn92**
**Subsystem**
gRPC client and server (`grpc-client`, `grpc-server`, and `grpc-core`)

**Problem Description**
The shared gRPC API does not expose several connection and call settings supported by grpc-java and gRPC C-core. Users need the same configuration surface on JVM and Native for keepalive, connection lifetime, inbound size limits, user agents, and wait-for-ready calls.

No YouTrack issue. See [discussion #748](https://github.com/Kotlin/kotlinx-rpc/discussions/748) for context.

**Solution**
This PR adds the missing client and server configuration options to the shared API. The JVM implementation delegates to the grpc-java builders, while the Native implementation sets the corresponding gRPC C-core channel arguments and call flags.

JVM and Native integration tests cover the new behavior. The PR also updates the generated API dumps and documents each option and its default.

**How to Verify**

- `./gradlew :grpc:grpc-core:jvmTest`
- `./gradlew :grpc:grpc-core:linuxX64Test`
- `./gradlew checkLegacyAbi detekt`
- `./gradlew :jpms-check:compileJava dokkaGenerate`

I ran these checks with JDK 21.

**Risk**
Low. The PR adds public APIs and leaves existing APIs unchanged. JVM delegates to existing grpc-java builder methods, and Native uses the matching gRPC C-core arguments. Integration tests cover both platform mappings.

**AI Assistance**
AI was used to verify the changes and check edge cases.

**YT Ticket Reference**
Fixes: KXRPC-641
Fixes: KXRPC-651

Referneces: KXRPC-645, KXRPC-643

### CODE REVIEW

**Comment by Mr3zee** [comment]
@nickjn92 thanks! I'll take a look at it soon!

File: grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcCallOptions.kt [hunk 1]
```diff
@@ -27,6 +27,18 @@ public class GrpcCallOptions {
...
+     * Whether the call should wait for a connection to become available instead of failing immediately.
+     *
+     * If `null` (the default), the channel's default behavior applies, which a service config delivered
+     * by the name resolver may override per method. If `true`, the call waits for the connection.
+     * If `false`, the call fails immediately when no connection is available, even if a service config
+     * enables waiting for this method.
+     *
+     * A deadline should generally be configured with [timeout] to prevent the call from waiting indefinitely.
+     */
+    public var waitForReady: Boolean? = null
```
Comment (Mr3zee) [inline grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcCallOptions.kt:40]: Please add the link to the semantics doc: https://github.com/grpc/grpc/blob/master/doc/wait-for-ready.md

File: grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcClient.kt [hunk 2]
```diff
@@ -271,6 +273,42 @@ public class GrpcClientConfiguration internal constructor() {
      */
     public var userAgent: String? = null
 
+    /**
+     * The maximum size, in bytes, of an inbound message accepted by this client.
+     *
+     * If `null` (the default), the gRPC runtime's default limit is used. Values must be non-negative.
```
Comment (Mr3zee) [inline grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcClient.kt:279]: Please list defaults per platform, in docs, and for other methods with defaults

**Review by Mr3zee — CHANGES_REQUESTED** [review]
@nickjn92 very nice! Thank you for contribution and sorry for the wait. I posted couple of nit picks before we proceed

Also, I'd like @Jozott00 to take a quick look into the native part, please

File: grpc/grpc-server/src/commonMain/kotlin/kotlinx/rpc/grpc/server/GrpcServer.kt [hunk 3]
```diff
@@ -198,6 +272,40 @@ public class GrpcServerConfiguration internal constructor() {
...
+    /**
+     * Server-side keepalive settings.
+     *
+     * @property time The time without read activity before a keepalive ping is sent.
+     * @property timeout The time to wait for read activity after sending a keepalive ping.
+     */
+    public class KeepAlive internal constructor() {
+        public var time: Duration = 2.hours
+            set(value) {
+                value.validateConnectionDuration("keepalive time", 1.milliseconds)
```
Comment (Mr3zee) [inline grpc/grpc-server/src/commonMain/kotlin/kotlinx/rpc/grpc/server/GrpcServer.kt:299]: It seems like the spec says it can't be lower than 10 seconds

File: docs/pages/kotlinx-rpc/topics/grpc-server.topic [hunk 4]
```diff
@@ -55,7 +55,49 @@
...
+                Without it, active RPCs may finish without a deadline.
+            </def>
         </deflist>
+        <note>
+            On the JVM, <code>keepAlive</code>, <code>maxConnectionIdle</code>,
+            <code>maxConnectionAge</code>, and <code>maxConnectionAgeGrace</code> require a server
+            provider that supports them. The bundled Netty provider does.
+        </note>
+        <note>
+            Duration options accept <code>Duration.INFINITE</code> to disable the behavior.
```
Comment (Mr3zee) [inline docs/pages/kotlinx-rpc/topics/grpc-server.topic:91]: A bit inaccurate for `keepAlive.timeout` and `maxConnectionAgeGrace`. It doesn't disable them, just makes infinite

File: grpc/grpc-server/src/nativeMain/kotlin/kotlinx/rpc/grpc/server/internal/NativeServer.kt [hunk 5]
```diff
@@ -348,6 +380,34 @@ internal class NativeServer(
...
+    maxInboundMessageSize?.let { add(GrpcArg.Integer(GRPC_ARG_MAX_RECEIVE_MESSAGE_LENGTH, it)) }
+    maxInboundMetadataSize?.let {
+        add(GrpcArg.Integer(GRPC_ARG_MAX_METADATA_SIZE, it))
+        add(GrpcArg.Integer(GRPC_ARG_ABSOLUTE_MAX_METADATA_SIZE, it))
+    }
+    keepAliveTime?.let { add(GrpcArg.Integer(GRPC_ARG_KEEPALIVE_TIME_MS, it.toChannelArgMilliseconds())) }
+    keepAliveTimeout?.let { add(GrpcArg.Integer(GRPC_ARG_KEEPALIVE_TIMEOUT_MS, it.toChannelArgMilliseconds())) }
+    maxConnectionIdle?.let { add(GrpcArg.Integer(GRPC_ARG_MAX_CONNECTION_IDLE_MS, it.toChannelArgMilliseconds())) }
+    maxConnectionAge?.let { add(GrpcArg.Integer(GRPC_ARG_MAX_CONNECTION_AGE_MS, it.toChannelArgMilliseconds())) }
+    maxConnectionAgeGrace?.let {
```
Comment (Mr3zee) [inline grpc/grpc-server/src/nativeMain/kotlin/kotlinx/rpc/grpc/server/internal/NativeServer.kt:406]: GRPC_ARG_KEEPALIVE_PERMIT_WITHOUT_CALLS is forgotten here

**Review by Mr3zee — COMMENTED** [review]
Some more comments:

1. This is still present in code even though other verification were introduced
```kotlin
keepAlive?.run {
            require(time.isPositive()) { "keepalive time must be positive" }
            require(timeout.isPositive()) { "keepalive timeout must be positive" }
        }
```

2. `assertMetadataSizeStatus` might capture unrelated failures sue to broad exceptions capture
3.  None of the three desktopNative actuals assert that configuration actually reaches the native server, is there any way to test it?

File: docs/pages/kotlinx-rpc/topics/grpc-server.topic [hunk 6]
```diff
@@ -55,7 +55,49 @@
...
+            Finite values must be under <code>Int.MAX_VALUE</code> milliseconds (about 24.8 days).
+            On Kotlin/Native these become gRPC channel arguments, whose values are a C
```
Comment (Jozott00) [inline docs/pages/kotlinx-rpc/topics/grpc-server.topic:93]: I think this sentence is too much implementation detail.

File: docs/pages/kotlinx-rpc/topics/grpc-server.topic [hunk 7]
```diff
@@ -55,7 +55,49 @@
...
+            <code>int</code>. <code>Duration.INFINITE</code> is passed as
+            <code>Int.MAX_VALUE</code>, the largest value such an argument can hold and the one
+            gRPC treats as effectively unbounded, so a finite value must stay below it to remain
+            distinguishable. The bound is enforced on every platform so that the same
+            configuration behaves identically everywhere; out-of-range values throw
+            <code>IllegalArgumentException</code>.
```
Comment (Jozott00) [inline docs/pages/kotlinx-rpc/topics/grpc-server.topic:99]: I think this should be fused with the first sentence (it repeats that the value must be smaller `Int.MAX_VALUE`).

Also I would remove "The bound is enforced on every platform so that the same configuration behaves identically everywhere;" which is expected.

File: grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcClient.kt (hunk 2 above)
Comment (Jozott00) [reply grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcClient.kt:279]: Maybe we should overwrite the runtime's defaults by our own one (probably one of the defaults)

File: grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcClient.kt [hunk 8]
```diff
@@ -271,6 +273,42 @@ public class GrpcClientConfiguration internal constructor() {
...
+    public var maxInboundMessageSize: Int? = null
+        set(value) {
+            require(value == null || value >= 0) { "maxInboundMessageSize must be >= 0" }
+            field = value
+        }
+
+    /**
+     * The maximum size, in bytes, of inbound metadata accepted by this client.
+     *
+     * If `null` (the default), the gRPC runtime's default limit is used. Values must be positive.
```
Comment (Jozott00) [inline grpc/grpc-client/src/commonMain/kotlin/kotlinx/rpc/grpc/client/GrpcClient.kt:290]: Same as above

**Review by Jozott00 — APPROVED** [review]
Native side looks good to me! 
Just nitpicks on the docs.

**Comment by Mr3zee** [comment]
Hey @nickjn92 ! Do you want to continue the work on this PR?
